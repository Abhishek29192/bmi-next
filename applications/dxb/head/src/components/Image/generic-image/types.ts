export interface Data {
  altText: string;
  src: string;
  width: number;
  height: number;
}

type Options = {
  className?: string;
  size?: "cover" | "contain";
  loading?: "lazy" | "eager";
  "data-testid"?: string;
};

export type Props = Data & Options;
