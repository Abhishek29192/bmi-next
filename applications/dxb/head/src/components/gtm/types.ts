export type DataLayer = Record<string, unknown>[];

declare global {
  interface Window {
    dataLayer: DataLayer;
    [key: string]: DataLayer;
  }
}

export type GTMParams = {
  gtmIds: string[];
  dataLayer?: string[];
  dataLayerName?: string;
  auth?: string;
  preview?: string;
  includeInDevelopment?: boolean;
  enableWebVitalsTracking?: boolean;
  selfHostedOrigin?: string;
  defaultDataLayer?: { platform: string; env: string };
};

export type GAParams = {
  gaId: string;
  dataLayerName?: string;
};
