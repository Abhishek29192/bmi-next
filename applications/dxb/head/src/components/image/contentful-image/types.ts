import { ImageWidths } from "../types";

type ImageData = {
  file: {
    fileName: string;
    contentType:
      | "image/jpg"
      | "image/jpeg"
      | "image/png"
      | "image/webp"
      | "image/gif"
      | "image/svg+xml";
    url: string;
    details: {
      image: {
        width: number;
        height: number;
      };
      size: number;
    };
  };
};

export interface Data {
  __typename: "ContentfulImage";
  altText: string;
  image: ImageData;
  title: string;
  type?: "Decorative" | "Descriptive" | null;
  focalPoint?: {
    x: number;
    y: number;
  } | null;
}

export type Options = {
  className?: string;
  size?: "cover" | "contain";
  position?: string;
  isMobile?: boolean;
  loading?: "lazy" | "eager";
  "data-testid"?: string;
};

export type Props = Omit<Data, "image"> & {
  widths: ImageWidths;
  image: { file: { url: ImageData["file"]["url"] } };
} & Options;
