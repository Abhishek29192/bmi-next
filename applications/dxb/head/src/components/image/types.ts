import type { GatsbyImageProps, IGatsbyImageData } from "gatsby-plugin-image";

type ImageData =
  | {
      file: {
        fileName: string;
        url?: never;
      };
      gatsbyImageData: IGatsbyImageData;
      thumbnail?: IGatsbyImageData;
    }
  | {
      file: {
        fileName: string;
        url: string;
      };
      gatsbyImageData?: never;
      thumbnail?: IGatsbyImageData;
    };

export type Data = {
  altText: string;
  type?: "Decorative" | "Descriptive" | null;
  image: ImageData;
  focalPoint?: {
    x: number;
    y: number;
  } | null;
  loading?: GatsbyImageProps["loading"];
};

export type Options = {
  className?: string;
  size?: "cover" | "contain";
  position?: string;
  isMobile?: boolean;
};

export type Props = Data & Options & { "data-testid"?: string };
