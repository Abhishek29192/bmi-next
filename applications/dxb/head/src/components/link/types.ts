import type { ButtonProps } from "@bmi-digital/components/button";
import type { GTM } from "@bmi-digital/components/types";
import type { IconName } from "../Icon";
import type { Data as PromoData } from "../Promo";
import type { SectionData } from "../Sections";
import type { Data as PageInfoData } from "../PageInfo";

type CustomisableButtonProps = Pick<
  ButtonProps,
  "startIcon" | "endIcon" | "variant" | "hasDarkBackground"
>;

export type LinkProps = {
  to: string;
  hasBrandColours?: boolean;
  className?: string;
  gtm?: Partial<GTM>;
  children: string;
};

export type ButtonLinkLinkProps = {
  to: string;
  hasBrandColours?: boolean;
  className?: string;
  gtm?: Partial<GTM>;
  children: string;
} & CustomisableButtonProps;

export type ToolProps = {
  parameters: Data["parameters"];
  onClick: (...args: unknown[]) => void;
  hasBrandColours?: boolean;
  className?: string;
  children: string;
  gtm?: Partial<GTM>;
};

export type ButtonLinkToolProps = {
  parameters: Data["parameters"];
  onClick: (...args: unknown[]) => void;
  hasBrandColours?: boolean;
  className?: string;
  children: string;
  gtm?: Partial<GTM>;
} & CustomisableButtonProps;

export type DialogProps = {
  dialogContent: Data["dialogContent"];
  onClick: (...args: unknown[]) => void;
  hasBrandColours?: boolean;
  className?: string;
  children: string;
  gtm?: Partial<GTM>;
};

export type ButtonLinkDialogProps = {
  dialogContent: Data["dialogContent"];
  onClick: (...args: unknown[]) => void;
  hasBrandColours?: boolean;
  className?: string;
  children: string;
  gtm?: Partial<GTM>;
} & CustomisableButtonProps;

export type HubSpotProps = {
  hubSpotCtaId: string;
  hasBrandColours?: boolean;
  className?: string;
  gtm?: Partial<GTM>;
  children: string;
};

export type ButtonLinkHubSpotProps = {
  hubSpotCtaId: string;
  hasBrandColours?: boolean;
  className?: string;
  gtm?: Partial<GTM>;
  children: string;
} & CustomisableButtonProps;

type BaseProps = {
  data: Data;
  children: string;
  hasBrandColours?: boolean;
  className?: string;
  gtm?: Partial<GTM>;
  onClick?: (...args: unknown[]) => void;
};

export type ButtonLinkProps = BaseProps & CustomisableButtonProps;

export type Props = BaseProps;

export enum DataTypeEnum {
  External = "External",
  Internal = "Internal",
  Asset = "Asset",
  Visualiser = "Visualiser",
  Calculator = "Calculator",
  Dialog = "Dialog",
  HubSpotCta = "HubSpot CTA"
}

export type Data = {
  __typename: "Link";
  id: string;
  label: string;
  icon: IconName | null;
  isLabelHidden: boolean | null;
  url: string | null;
  type: DataTypeEnum;
  parameters: { [key: string]: unknown } | null;
  dialogContent: SectionData | null;
  linkedPage: {
    path: string | null;
  } | null;
  asset?: {
    url: string | null;
  } | null;
  hubSpotCTAID: string | null;
  queryParams: string | null;
};

export type NavigationItem = {
  __typename: "NavigationItem";
  type: "Heading" | "Separator";
  value: string;
};

export type NavigationData = {
  __typename: "Navigation";
  label: string | null;
  link: Data | null;
  promos?: (PromoData | PageInfoData)[] | null;
  links: (NavigationData | NavigationItem | Data)[];
};
