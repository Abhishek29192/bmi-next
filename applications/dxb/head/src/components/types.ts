export const All_FORMATS = [
  "application/pdf",
  "application/zip",
  "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
  "image/jpg",
  "image/jpeg",
  "image/png",
  "image/svg+xml",
  "text/plain",
  "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
] as const;
export type Format = (typeof All_FORMATS)[number];
