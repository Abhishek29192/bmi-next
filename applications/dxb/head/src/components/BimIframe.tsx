import React from "react";
import { Asset } from "./types/ProductBaseTypes";
import styles from "./styles/BimIframe.module.scss";

type Props = {
  url: string;
};

export const getBimIframeUrl = (assets?: ReadonlyArray<Asset>): string | null =>
  assets?.find((asset) => asset.assetType === "BIM")?.url || null;

const BimIframe = ({ url }: Props) => (
  <iframe className={styles["bimIframe"]} src={url} />
);

export default BimIframe;
