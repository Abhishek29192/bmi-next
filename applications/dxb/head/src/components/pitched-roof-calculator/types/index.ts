export type RangeValue = {
  start: number;
  end: number;
  value: number;
};

export type BaseProduct = {
  code: string;
  name: string;
};

export type BaseVariant = BaseProduct & {
  externalProductCode: string;
  image: string;
};

export type LengthBasedProduct = BaseVariant & {
  length: number;
};

export type WidthBasedProduct = BaseVariant & {
  width: number;
};

export type VergeMetalFlushOption = {
  type: "METAL_FLUSH";
  name: string;
  left: LengthBasedProduct;
  right: LengthBasedProduct;
  leftStart?: LengthBasedProduct;
  rightStart?: LengthBasedProduct;
};

export type VergeTileOption = {
  type: "TILE";
  name: string;
  left: WidthBasedProduct;
  right: WidthBasedProduct;
  halfLeft: WidthBasedProduct;
  halfRight: WidthBasedProduct;
};

export type VergeOption = VergeTileOption | VergeMetalFlushOption;

export type Accessory = BaseVariant & {
  category: ProductCategory;
  packSize?: number;
};

export type Underlay = BaseVariant & {
  description?: string;
  minSupportedPitch: number;
  length: number;
  width: number;
  overlap: number;
};

export type GutteringVariant = LengthBasedProduct & {
  downpipe: Accessory;
  downpipeConnector: Accessory;
};

export type Guttering = BaseProduct & {
  image: string;
  variants: GutteringVariant[];
};

export type BaseTile = {
  minBattenGauge: number;
  maxBattenGauge: RangeValue[];
  eaveGauge: RangeValue[];
  ridgeSpacing: RangeValue[];
  width: number;
  height: number;
  brokenBond: boolean;
};

export type MainTileVariant = BaseVariant &
  BaseTile & {
    color: string;
    halfTile?: WidthBasedProduct;
    hip: LengthBasedProduct;
    ridgeOptions: LengthBasedProduct[];
    vergeOptions: VergeOption[];
    valleyMetalFlushStart?: LengthBasedProduct;
    valleyMetalFlush?: LengthBasedProduct;
    valleyMetalFlushEnd?: LengthBasedProduct;
    valleyMetalFlushTop?: LengthBasedProduct;
    valleyMetalFlushDormerStart?: LengthBasedProduct;
    accessories: Accessory[];
    eaveAccessories: Accessory[];
    clip?: Accessory;
    ridgeAndHipScrew?: Accessory;
    longScrew?: Accessory;
    screw?: Accessory;
    stormBracket?: Accessory;
    finishingKit?: Accessory;
    ventilationHoodOptions: Accessory[];
  };

export type MainTileCategory = "concrete" | "metal" | "clay";

export type MainTile = BaseProduct &
  BaseTile & {
    category: MainTileCategory;
    variants: MainTileVariant[];
  };

export enum ProductCategory {
  Tiles = "tiles",
  Sealing = "sealing",
  Fixings = "fixings",
  Ventilation = "ventilation",
  Accessories = "accessories"
}

export type Data = {
  mainTiles: MainTile[];
  gutters: Guttering[];
  gutterHooks: LengthBasedProduct[];
  underlays: Underlay[];
};

export type ResultsRow = {
  category: ProductCategory;
  image: string;
  description: string;
  externalProductCode: string;
  packSize: string;
  quantity: number;
};

export type ResultsObject = {
  [category in ProductCategory]: ResultsRow[];
};

export enum CalculatorSteps {
  SelectRoof = "select-roof",
  EnterDimensions = "enter-dimensions",
  SelectTile = "select-tile",
  SelectVariant = "select-variant",
  TileOptions = "tile-options",
  SelectUnderlay = "select-underlay",
  Guttering = "guttering",
  YourSolutionContains = "your-solution-contains"
}

export type CalculatorConfig = {
  hubSpotFormId: string | null;
  roofShapes: { roofShapeId: string }[];
};
