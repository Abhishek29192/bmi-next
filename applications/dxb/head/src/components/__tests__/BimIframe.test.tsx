import React from "react";
import { render } from "@testing-library/react";
import { Asset } from "../types/ProductBaseTypes";
import BimIframe, { getBimIframeUrl } from "../BimIframe";

describe("BimIframe component", () => {
  it("renders correctly", () => {
    const { container } = render(<BimIframe url="https://google.com" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("getBimIframeUrl", () => {
  const allAssets: Asset[] = [
    {
      assetType: "BIM",
      name: "BIM URL Sample",
      realFileName: "foo",
      url: "https://www.bimobject.com/fr/bmi_siplast_fr/product/spl010108"
    },
    {
      assetType: "WARRANTIES",
      name: "Garanti Test",
      realFileName: "Monier garanti 30 Ã¥r.jpg",
      url: "https://bmipimngqa.azureedge.net/sys-master-hybris-media/hc2/h3e/8972787286046/Monier-garanti-30-aarjpg"
    },
    {
      assetType: "AWARDS",
      name: "test award",
      realFileName: "Test_Award.png",
      url: "https://bmipimngqa.azureedge.net/sys-master-hybris-media/h03/h62/9006918271006/Test-Awardpng"
    },
    {
      assetType: "GUARANTIES",
      name: "Test Guarentee (NO)",
      realFileName: "Test_Guarantee.png",
      url: "https://bmipimngqa.azureedge.net/sys-master-hybris-media/h05/h5f/9006918336542/Test-Guaranteepng"
    },
    {
      assetType: "CERTIFICATES",
      name: "Test Certificate (NO)",
      realFileName: "test_Certificate.png",
      url: "https://bmipimngqa.azureedge.net/sys-master-hybris-media/hf4/h5b/9006918402078/test-Certificatepng"
    }
  ];

  it("no assets", () => {
    const assets = null;

    const result = getBimIframeUrl(assets);

    expect(result).toBe(null);
  });

  it("not found url", () => {
    const assets: Asset[] = [allAssets[1], allAssets[2]];

    const result = getBimIframeUrl(assets);

    expect(result).toBe(null);
  });

  it("found url", () => {
    const assets: Asset[] = [...allAssets];

    const result = getBimIframeUrl(assets);

    expect(result).toBe(assets[0].url);
  });
});
