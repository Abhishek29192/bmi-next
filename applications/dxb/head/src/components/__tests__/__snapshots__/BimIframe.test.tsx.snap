// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BimIframe component renders correctly 1`] = `
<div>
  <iframe
    class="bimIframe"
    src="https://google.com"
  />
</div>
`;
